### PR TITLE
DOC: Tidy wsicore Docstrings

### PR DESCRIPTION
**Pull request by John-P**
- Make line wrapping consistent
- Make indentation consistent

### CODE REVIEW

File: tiatoolbox/wsicore/wsireader.py [hunk 1]
```diff
@@ -479,12 +494,15 @@ def slide_dimensions(
...
+            units (str):
+                resolution units, default="power".
 
         Returns:
-            :py:obj:`tuple`: shape of WSI in (width, height).
+            :py:obj:`tuple`:
+                Size of the WSI in (width, height).
 
         Examples:
             >>> from tiatoolbox.wsicore import wsireader
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:508]:
```suggestion
            >>> from tiatoolbox.wsicore.wsireader import WSIReader
            >>> wsi = WSIReader.open()
```

File: tiatoolbox/wsicore/wsireader.py [hunk 2]
```diff
@@ -548,29 +570,35 @@ def _find_read_bounds_params(
     def convert_resolution_units(self, input_res, input_unit, output_unit=None):
         """Converts resolution value between different units.
 
-        This function accepts a resolution and its units in the input and converts
-        it to all other units ('mpp', 'power', 'baseline'). To achieve resolution
-        in 'mpp' and 'power' units in theoutput, WSI meta data should cotain `mpp`
-        and `objective_power` information, respectively.
+        This function accepts a resolution and its units in the input
+        and converts it to all other units ('mpp', 'power', 'baseline').
+        To achieve resolution in 'mpp' and 'power' units in theoutput,
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:575]:
```suggestion
        To achieve resolution in 'mpp' and 'power' units in the output,
```

File: tiatoolbox/wsicore/wsireader.py [hunk 3]
```diff
@@ -728,54 +756,61 @@ def read_rect(
...
-                VirtualWSIReader. See class docstrings for more
-                information.
+            coord_space (str):
+                Defaults to "baseline". This is a flag to indicate if
+                the input `bounds` is in the baseline coordinate system
+                ("baseline") or is in the requested resolution system
+                ("resolution").
+            **kwargs (dict):
+                Extra key-word arguments for reader specific parameters.
+                Currently only used by VirtualWSIReader. See class
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:808]:
```suggestion
                Currently, only used by VirtualWSIReader. See class
```

File: tiatoolbox/wsicore/wsireader.py [hunk 4]
```diff
@@ -728,54 +756,61 @@ def read_rect(
...
+                docstrings for more information.
 
         Returns:
-            :class:`numpy.ndarray`: array of size MxNx3 M=size[0], N=size[1]
+            :class:`numpy.ndarray`:
+                Array of size MxNx3 M=size[0], N=size[1]
 
         Example:
             >>> from tiatoolbox.wsicore.wsireader import get_wsireader
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:816]:
```suggestion
            >>> from tiatoolbox.wsicore.wsireader import WSIReader
            >>> WSIReader.open()
```

File: tiatoolbox/wsicore/wsireader.py [hunk 5]
```diff
@@ -806,27 +841,27 @@ def read_rect(
...
-        of the requested region are projected to the next highest
-        resolution. This resolution is then decoded and downsampled
-        to produced the desired output. This is a major source of
-        variability in the time take to perform a read operation. Reads
-        which require reading a large region before downsampling will
-        be significantly slower than reading at a fixed level.
+        When reading between the levels stored in the WSI, the
+        coordinates of the requested region are projected to the next
+        highest resolution. This resolution is then decoded and
+        downsampled to produced the desired output. This is a major
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:860]:
```suggestion
        downsampled to produce the desired output. This is a major
```

File: tiatoolbox/wsicore/wsireader.py [hunk 6]
```diff
@@ -1034,15 +1074,19 @@ def read_region(self, location: NumPair, level: int, size: IntPair) -> np.ndarra
...
+            units (str):
+                Resolution units, default="power".
 
         Returns:
-            :class:`numpy.ndarray`: thumbnail image
+            :class:`numpy.ndarray`:
+                Thumbnail image.
 
         Examples:
             >>> from tiatoolbox.wsicore.wsireader import get_wsireader
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:1092]:
```suggestion
            >>> from tiatoolbox.wsicore.wsireader import WSIReader
            >>> WSIReader.open()
```

File: tiatoolbox/wsicore/wsireader.py [hunk 7]
```diff
@@ -2521,16 +2585,17 @@ def get_wsireader(input_img):
...
+          tiatoolbox WSI handler. In the latter case, the function
+          directly passes the input_imge to the output.
 
     Returns:
-        WSIReader: an object with base :class:`.WSIReader` as base class.
+        WSIReader:
+            An object with base :class:`.WSIReader` as base class.
 
     Examples:
         >>> from tiatoolbox.wsicore.wsireader import get_wsireader
```
Comment (shaneahmed) [inline tiatoolbox/wsicore/wsireader.py:2601]:
```suggestion
        >>> from tiatoolbox.wsicore.wsireader import WSIReader
        >>> WSIReader.open()
```

**Review by shaneahmed — COMMENTED** [review]
Thanks @John-P 
Please address the comments before we can merge.

File: tiatoolbox/wsicore/wsireader.py [hunk 8]
```diff
@@ -548,29 +570,35 @@ def _find_read_bounds_params(
...
+            output_res (float or dictionary):
+                Either a float which is the converted `input_res` to the
+                desired `output_unit` or a dictionary containing the
+                converted `input_res` to all acceptable units (`'mpp'`,
+                `'power'`, `'baseline'`). If there is not enough meta
+                data to calculate a unit (like `mpp` or `power`), they
+                will be set to None in the dictionary.
 
         """
         baseline_mpp = self.info.mpp
```
[the remark this answers is not in the document]
Comment (John-P) [reply tiatoolbox/wsicore/wsireader.py:605]: I am not sure what you mean. Can you give a permalink? The line numbers will change between commits.

**Comment by John-P** [comment]
Thanks @shaneahmed, I have addressed your comments and suggestions.

File: tiatoolbox/wsicore/wsireader.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (shaneahmed) [reply tiatoolbox/wsicore/wsireader.py:604]: Fixed now https://github.com/TissueImageAnalytics/tiatoolbox/blob/0a0963fe7734856819d222195866fdf660d975cd/tiatoolbox/wsicore/wsireader.py#L663

**Review by shaneahmed — APPROVED** [review]

**Pull request merged**